### PR TITLE
Finish Chapter 2

### PR DESCRIPTION
**Pull request by hasrthur**
### Solutions for Chapter 2

cc @vrom911 @chshersh

Thanks for your time

### CODE REVIEW

File: src/Chapter2.hs [hunk 1]
```diff
@@ -349,7 +351,10 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = take halfLength l
+  where
+    halfLength = div (length l) 2
```
Comment (vrom911) [inline src/Chapter2.hs:357]: Great usage of `where` construction! Makes code easily readable 💯

File: src/Chapter2.hs [hunk 2]
```diff
@@ -620,7 +628,9 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven (x:_:xs) = x:takeEven xs
```
Comment (vrom911) [inline src/Chapter2.hs:632]: Great pattern matching skills 🔥

File: src/Chapter2.hs [hunk 3]
```diff
@@ -780,13 +791,15 @@ Let's now try to eta-reduce some of the functions and ensure that we
...
 
 -- TODO: type ;)
-listElementsLessThan x l = filter (< x) l
+listElementsLessThan :: Int -> [Int] -> [Int]
+listElementsLessThan x = filter (< x)
 
 -- Can you eta-reduce this one???
-pairMul xs ys = zipWith (*) xs ys
+pairMul :: [Int] -> [Int] -> [Int]
+pairMul = zipWith (*)
```
Comment (vrom911) [inline src/Chapter2.hs:802]: Eta-reduction award 🏆

**Review by vrom911 — APPROVED** [review]
Perfectly done 💠 
Congrats with finishing Chapter Two!
I really enjoyed your solutions 💯  And fantastic job with the advanced task!

This PR is ready to go, feel free to merge it right away 🚢

**Comment by hasrthur** [comment]
> Perfectly done 💠
> Congrats with finishing Chapter Two!
> I really enjoyed your solutions 💯 And fantastic job with the advanced task!
> 
> This PR is ready to go, feel free to merge it right away 🚢

Appreciate your time!

**Pull request merged**